Make January 2024 news public (#26679)
idaholab/moose: modules/doc/content/newsletter/2024/2024_01.md

# MOOSE Newsletter (January 2024)

## MOOSE Improvements

### New Physics action syntax

> The [Physics](syntax/Physics/index.md) syntax has been added to MOOSE. The `Physics` system is meant
> to streamline the input of an equation in MOOSE, by selecting good default parameters and preconditioners
> for example. `Physics` are derived from [Actions](syntax/Actions/index.md) but define additional APIs
> to facilitate the implementation of a new `Physics` action. As appropriate, some of the modules' actions that defined equations will be transitioned to the `Physics` system over the coming months.

Author: ```suggestion
The [Physics](syntax/Physics/index.md) syntax has been added to MOOSE. The `Physics` system is meant
to streamline the input of an set of physical equations in MOOSE. For example, this might allow the definition of the required equations automatically but also the selection of good default parameters and preconditioners. `Physics` are derived from [Actions](syntax/Actions/index.md) but define additional APIs
to facilitate the implementation of a new `Physics` action. As appropriate, some of the modules' actions that defined equations will be transitioned to the `Physics` system over the coming months.
```

@GiudGiud

### Separation of the Materials and FunctorMaterials block

[Functor materials](syntax/FunctorMaterials/index.md) were added to MOOSE to
allow evaluation of material properties at arbitrary locations, such as element

> faces or nodes. These materials do not define material properties in the traditional
> [Material](syntax/Materials/index.md) material system, but instead they define
> [Functors](syntax/Functors/index.md). To mark the difference between the two system,
> the two types of objects should no longer be created in the same `[Materials]` block,
> FunctorMaterials should be created inside a `[FunctorMaterials]` block.

Author: ```suggestion
faces or nodes. These materials do not define traditional material properties as in the
[Material](syntax/Materials/index.md) system, but instead they define
[Functors](syntax/Functors/index.md). To mark the difference between the two systems,
the two types of objects should no longer be created in the same `[Materials]` block; 
FunctorMaterials should be created inside a `[FunctorMaterials]` block.
```

@GiudGiud

### Improvements in the Navier Stokes Module

> - k-$\epsilon$ turbulence model was added. This method enables more accurate turbulence modeling compared to the current mixing-length model.
> - A drift-flux mixture model was added which enables solutions of two-phase flows where the phases move at different velocities, but assume local equilibrium over short spatial length scales.

Author: ```suggestion
- The k-$\epsilon$ turbulence model was added. This method enables more accurate turbulence modeling compared to the current mixing-length model.
- A drift-flux mixture model was added which enables solutions of two-phase flows where the phases move at different velocities but assume local equilibrium over short spatial length scales.
```

@grmnptr

# MOOSE Newsletter (January 2024)

## MOOSE Improvements

### New Physics action syntax

The [Physics](syntax/Physics/index.md) syntax has been added to MOOSE. The `Physics` system is meant

> to streamline the input of an set of physical equations in MOOSE. For example, this might allow the

Reviewer: ```suggestion
to streamline the input of a set of physical equations in MOOSE. For example, this might allow the
```

definition of the required equations automatically but also the selection of good default parameters

> and preconditioners. `Physics` are derived from [Actions](source/actions/Action.md) but define additional

Reviewer: ```suggestion
and preconditioners. `Physics` are derived from [Actions](source/actions/Action.md) but define
```

APIs to facilitate the implementation of a new `Physics` action. As appropriate, some of the modules'
actions that defined equations will be transitioned to the `Physics` system over the coming months.

### Separation of the Materials and FunctorMaterials block

[Functor materials](syntax/FunctorMaterials/index.md) were added to MOOSE to
allow evaluation of material properties at arbitrary locations, such as element

> faces or nodes. These materials do not define traditional material properties as in the

Reviewer: ```suggestion
faces or nodes, usually on-the-fly (although some caching is supported). These materials do not define traditional material properties as in the
```

## PETSc-level Changes

## Bug Fixes and Minor Enhancements

- Shorthand functor material property output was added using the [!param](/FunctorMaterials/GenericFunctorMaterial/output_properties) parameter. This automatically creates a [FunctorAux.md] and a constant monomial variable for output, which had to be done manually before.
- Extra element IDs can now be used to assign elements to subdomains in the [ParsedSubdomainMeshGenerator.md].
- The [PropertyReadFile.md] user object can now read multiple files, once every time it is executed. The execution is controlled with the [!param](/UserObjects/PropertyReadFile/execute_on) parameter.

> - The [NEML2](syntax/NEML2/index.md) constitutive model library was added to the tensor mechanics module as an optional dependency. NEML2 is built on top of libTorch and supports teh implementation of devide independent material models (with GPU support).

Author: ```suggestion
- The [NEML2](syntax/NEML2/index.md) constitutive model library was added to the tensor mechanics module as an optional dependency. NEML2 is built on top of libTorch and supports the implementation of devide independent material models (with GPU support).
```

@dschwen One more typo here I didn't know how to fix - what did you mean insteade of "devide"?
